disco/storage/ingesters.py

```diff
@@ -0,0 +1,365 @@
+        columns = ", ".join(columns)
+        conn = sqlite3.connect(self.database)
+        try:
+            sql = f"INSERT INTO {table} ({columns}) VALUES ({values})"
+            conn.executemany(sql, data)
+            conn.commit()
+        finally:
+            conn.close()
+        logger.info("Ingestion success - '%s'.", table)
+
+
+class TaskIngester(IngesterBase):
+    """Class used for ingesting parsed task into database"""
+    
+    data_class = Task
+
+    def ingest(self, task):
+        """Ingest task data into task table in database"""
+        columns = self.data_class.__table__.columns.keys()
+        data = [tuple([task[column] for column in columns])]
+        self._perform_ingestion(columns=columns, data=data)
+        indexes = {task["name"]: task["id"]}
+        return indexes
+
+
+class ScenarioIngester(IngesterBase):
+    """Class used for ingesting parsed scenarios into database"""
+    
+    data_class = Scenario
+    
+    def ingest(self, scenarios):
+        """Ingest a list of scenarios into scenario table in database"""
+        columns = self.data_class.__table__.columns.keys()
+        data = [tuple([item[column] for column in columns]) for item in scenarios]
+        self._perform_ingestion(columns=columns, data=data)
+        indexes = {self._generate_identifier(item): item["id"] for item in scenarios}
+        return indexes
+    
+    @staticmethod
+    def _generate_identifier(item):
+        fields = ["job_id", "simulation_type", "name"]
+        values = [item[k] for k in fields]
+        return " | ".join(values)
+
+
+class JobIngester(IngesterBase):
+    """Class used for ingesting parsed jobs into database"""
+    
+    data_class = Job
+    
+    def ingest(self, jobs):
+        """Ingest a list of jobs into job table in database"""
+        columns = self.data_class.__table__.columns.keys()
+        data = [tuple([item[column] for column in columns]) for item in jobs]
+        self._perform_ingestion(columns=columns, data=data)
+        indexes = {job["name"]: job["id"] for job in jobs}
+        return indexes
+
+
+class ReportIngester(IngesterBase):
+    """Class used for ingesting parsed reports into database"""
+    
+    data_class = Report
+
+    def ingest(self, reports):
+        """Ingest parsed reports into report table in database"""
+        columns = self.data_class.__table__.columns.keys()
+        data = [tuple([item[column] for column in columns]) for item in reports]
+        self._perform_ingestion(columns=columns, data=data)
+        indexes = {report["file_name"]: report["id"] for report in reports}
+        return indexes
+
+
+class TableIngesterMixin:
+    """Mixin class for handling report table ingestion"""
+
+    def ingest(self, objects):
+        columns = self.data_class.__table__.columns.keys()
+        data = [tuple([item[column] for column in columns]) for item in objects]
+        self._perform_ingestion(columns=columns, data=data)
```

Comment:
The attribute and method are from the parent class `IngesterBase`.

Reply:
It seems odd because there is an implicit requirement that this mixin only be in class that is also a subclass of IngesterBase. When looking only at this class, it appears to be an error. Do people often follow this pattern with mixins?

I'm ok leaving it as-is.